clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/userdata/LocalFileUserDataProviderSpec.groovy

```diff
@@ -17,42 +17,47 @@
 
 package com.netflix.spinnaker.clouddriver.aws.deploy.userdata
 
-import com.netflix.spinnaker.clouddriver.aws.deploy.userdata.LocalFileUserDataProperties
-import com.netflix.spinnaker.clouddriver.aws.deploy.userdata.LocalFileUserDataProvider
 import spock.lang.Specification
 
 class LocalFileUserDataProviderSpec extends Specification {
 
-  final String APP = 'app'
-  final String STACK = 'stack'
-  final String COUNTRIES = 'countries'
-  final String DEV_PHASE = 'devPhase'
-  final String HARDWARE = 'hardware'
-  final String PARTNERS = 'partners'
-  final String REVISION = 99
-  final String ZONE = 'zone'
-  final String REGION = 'region'
-  final String ACCOUNT = 'account'
-  final String ENVIRONMENT = 'environment'
-  final String ACCOUNT_TYPE = 'accountType'
+  static final String APP = 'app'
+  static final String STACK = 'stack'
+  static final String DETAIL = 'detail'
+  static final String COUNTRIES = 'countries'
+  static final String DEV_PHASE = 'devPhase'
+  static final String HARDWARE = 'hardware'
+  static final String PARTNERS = 'partners'
+  static final String REVISION = 99
+  static final String ZONE = 'zone'
+  static final String REGION = 'region'
+  static final String ACCOUNT = 'account'
+  static final String ENVIRONMENT = 'environment'
+  static final String ACCOUNT_TYPE = 'accountType'
 
-  final String ASG_NAME = "${APP}-${STACK}-c0${COUNTRIES}-d0${DEV_PHASE}-h0${HARDWARE}-p0${PARTNERS}-r0${REVISION}-z0${ZONE}"
-  final String LAUNCH_CONFIG_NAME = 'launchConfigName'
+  static final String ASG_NAME = "${APP}-${STACK}-${DETAIL}-c0${COUNTRIES}-d0${DEV_PHASE}-h0${HARDWARE}-p0${PARTNERS}-r0${REVISION}-z0${ZONE}"
+  static final String LAUNCH_CONFIG_NAME = 'launchConfigName'
 
   void "replaces expected strings"() {
     given:
     LocalFileUserDataProvider localFileUserDataProvider = GroovySpy()
     localFileUserDataProvider.localFileUserDataProperties = new LocalFileUserDataProperties()
-    localFileUserDataProvider.assembleUserData(_, _, _) >> getRawUserData()
+    localFileUserDataProvider.isLegacyUdf(_, _) >> legacyUdf
+    localFileUserDataProvider.assembleUserData(legacyUdf, _, _, _) >> getRawUserData()
 
     when:
     def userData = localFileUserDataProvider.getUserData(ASG_NAME, LAUNCH_CONFIG_NAME, REGION, ACCOUNT, ENVIRONMENT, ACCOUNT_TYPE)
 
     then:
-    userData == getFormattedUserData()
+    userData == getFormattedUserData(expectedEnvironment)
+
+    where:
+    legacyUdf | expectedEnvironment
+    true      | ACCOUNT
```

Comment:
in the legacy format we didn't have account and environment as separate concepts, so where we were templating environment we were really templating the account name